### PR TITLE
New follow functionality in profile component

### PR DESCRIPTION
**Pull request by TobiasAschenbrenner**
Implemented the following functionality. ([CHP-7](https://project-saving-twitter.atlassian.net/browse/CHP-7))

This PR includes:
- Added the "following" state to the Profile component, which stores the current following status of the displayed user.
- Created the "handleFollow" function to handle following and unfollowing users by making API requests.
- Implemented a hover effect for the follow button to switch between "Following" and "Unfollow" when the user is already following the displayed user.
- Updated the CSS for the "following-button" and "follower-button" classes to remove the background color and add an underline effect on hover.
- Modified the Profile component to fetch and display the list of followings and followers when the corresponding buttons are clicked.
- Created a new state variable "showList" to control the display of the followings or followers list.
- Created the "fetchFollowings" and "fetchFollowers" functions to fetch the followings and followers list from the API.
- Added event handlers to the "following-button" and "follower-button" elements to call the respective fetch functions and update the "showList" state when clicked.
- Implemented a modal to display the fetched list, with a darkened background.
- Centered the modal and styled the list elements for improved user experience.
- Added appropriate error handling and cleanup for the fetch requests and event listeners.

### CODE REVIEW

File: src/components/Profile/Profile.css
Comment (koenidv) [inline src/components/Profile/Profile.css:1]: Let's continue migrating towards scss :)

File: src/components/Profile/Profile.js [hunk 1]
```diff
@@ -1,21 +1,204 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import "./Profile.css";
 
+const API_BASE_URL = "https://api.chirp.koenidv.de";
+
+const getUsernameFromJWT = async () => {
+  const token = localStorage.getItem("jwt");
+
+  if (token) {
```
Comment (koenidv) [inline src/components/Profile/Profile.js:9]: Isn't this given by being surrounded by the SecureRoute component?

**Review by koenidv — APPROVED** [review]
overall, lgtm

File: src/components/Profile/Profile.js (hunk 1 above)
Comment (TobiasAschenbrenner) [reply src/components/Profile/Profile.js:9]: You're right it is. Removed it.

**Pull request merged**